src/js/requestHandlers.js

```
@@ -230,6 +235,83 @@ sjrk.storyTelling.server.deleteStoryFromCouch = function (id, deleteStoryDataSou
+        } else if (block.blockType === "audio" || block.blockType === "video") {
+            blockFileName = block.mediaUrl;
+        }
+
+        if (sjrk.storyTelling.server.isValidMediaFilename(blockFileName)) {
+            filesToDelete.push(blockFileName);
+        }
+    });
+
+    // remove duplicate entries so we don't try to delete already-deleted files
+    filesToDelete = filesToDelete.filter(function (fileName, index, self) {
+        return self.indexOf(fileName) === index;
+    });
+
+    fluid.each(filesToDelete, function (fileToDelete) {
+        sjrk.storyTelling.server.deleteSingleFileRecoverable(fileToDelete, serverConfig);
+    });
+};
+
+/*
+ * Verifies that a given file name follows the UUID format as laid out in
+ * RFC4122. A detailed description of the format can be found here:
+ * https://en.wikipedia.org/wiki/Universally_unique_identifier#Format
+ */
+sjrk.storyTelling.server.isValidMediaFilename = function (fileName) {
+    if (fileName && typeof fileName === "string" ) {
+        var fileNameWithoutExtension = path.basename(fileName, path.extname(fileName));
+
+        var validUuid = /\b[0-9a-f]{8}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{4}-[0-9a-f]{12}\b/;
+
+        return validUuid.test(fileNameWithoutExtension);
+    } else {
+        return false;
+    }
+};
+
+sjrk.storyTelling.server.getServerPathForFile = function (fileName, directoryName) {
+    return "." + directoryName + path.sep + fileName;
+};
+
+sjrk.storyTelling.server.deleteSingleFileRecoverable = function (fileToDelete, serverConfig) {
+    var recoveryPath = sjrk.storyTelling.server.getServerPathForFile(fileToDelete, serverConfig.deletedFilesRecoveryPath);
+    var deletionPath = sjrk.storyTelling.server.getServerPathForFile(fileToDelete, serverConfig.uploadedFilesHandlerPath);
+
+    // make sure the file is in the uploads dir before we begin
+    try {
+        fs.accessSync(deletionPath, fs.constants.W_OK | fs.constants.R_OK);
```

We don't need this test. Just let the rename fail.